src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -120,14 +121,23 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 				{
 					Constraint *constraint = (Constraint *) command->def;
 					char **constraintName = &(constraint->conname);
+					const bool missingOk = false;
+					relationId = RangeVarGetRelid(alterTableStmt->relation,
+												  AccessShareLock,
+												  missingOk);
 
 					if (constraint->contype == CONSTR_PRIMARY && constraint->indexname)
 					{
 						char **indexName = &(constraint->indexname);
 						AppendShardIdToName(indexName, shardId);
 					}
 
-					AppendShardIdToName(constraintName, shardId);
+					/* append shardId to constraint names when necessary */
+					if (!PartitionedTable(relationId) ||
```

Comment:
What happens if the partition is on the primary key, why don't we have the same problem then?

Reply:
I failed to create a primary key constraint that includes both the partition and the distribution column after distributing the table.

Before distributing the table, the user can create a composite primary key easily.

    ALTER TABLE part_table ADD PRIMARY KEY (work_ymdt, seq);

However after distributing the table, the same query is not supported.

	ALTER TABLE part_table ADD PRIMARY KEY (work_ymdt, seq);
	ERROR:  0A000: cannot create constraint without a name on a distributed table

Creating a unique index and trying to use that index do not work as well

	CREATE UNIQUE INDEX uniq_work_ymdt_seq ON part_table (work_ymdt, seq);
	CREATE INDEX

	ALTER TABLE part_table ADD CONSTRAINT pk PRIMARY KEY USING INDEX uniq_work_ymdt_seq;
	ERROR:  0A000: ALTER TABLE / ADD CONSTRAINT USING INDEX is not supported on partitioned tables

Reply:
what if the distribution and partition key are the same, as in `seq` in this example

Reply:
> what if the distribution and partition key are the same, as in `seq` in this example

Then all the records in a shard will be in the same partition and the other partitions will always be empty. I do not see how this can be a valid distribution strategy.

Nonetheless, I will test and see how it behaves

Reply:
I tested and it works. I think this is a trivial thing to test. @SaitTalhaNisanci do you think it is worth adding a regression test for this? Can I resolve this conversation?

```sql
--1. create a partitioned table
CREATE TABLE part_table (
    seq bigint NOT NULL,
    CONSTRAINT seq_check CHECK (seq <= 200))
)
PARTITION BY RANGE (seq);

--2. perform create_distributed_table
SELECT create_distributed_table('part_table', 'seq');

--3. add a partition
CREATE TABLE part_table_p_1_100 PARTITION OF part_table  FOR VALUES FROM (1) TO (100);

--4. add a check constraint
alter table part_table add CONSTRAINT my_seq CHECK (my_seq > 0);

CREATE TABLE part_table_p_101_200 PARTITION OF part_table  FOR VALUES FROM (101) TO (200);
```

Reply:
Thanks for testing, I am resolving it Hanefi.